components/apps_form/apps_form_field/select_user_option.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {components, OptionProps} from 'react-select';
+
+import {AppSelectOption} from '@mattermost/types/apps';
+import Avatar from 'components/widgets/users/avatar/avatar';
+
+import * as Utils from 'utils/utils';
+import BotBadge from 'components/widgets/badges/bot_badge';
+import GuestBadge from 'components/widgets/badges/guest_badge';
+import {isGuest} from 'mattermost-redux/utils/user_utils';
+
+const {Option} = components;
+export const SelectUserOption = (props: OptionProps<AppSelectOption>) => {
+    const username = props.data.username;
+    let description = '';
+
+    if ((props.data.first_name || props.data.last_name) && props.data.nickname) {
+        description = ` - ${Utils.getFullName(props.data)} (${props.data.nickname})`;
+    } else if (props.data.nickname) {
+        description = ` - (${props.data.nickname})`;
+    } else if (props.data.first_name || props.data.last_name) {
+        description = ` - ${Utils.getFullName(props.data)}`;
+    }
```

Comment:
What's the behavior if we use `getFullName` unconditionally here?

Reply:
used this in accordance with how it was in AutocompleteSelector. should use `getFullName` only or the current one @mickmister ?

Reply:
It's probably fine to do what the `AutocompleteSelector` component was doing